### PR TITLE
fix(setup-wallet): regression issues

### PR DESCRIPTION
**Pull request by banklesss**
Take the opportunity to apply some refactors

Also related to:

- [YOMO-1402](https://emurgo.atlassian.net/browse/YOMO-1402) (comment)
- [YOMO-1351](https://emurgo.atlassian.net/browse/YOMO-1351) (comment)
- [YOMO-1353](https://emurgo.atlassian.net/browse/YOMO-1353) (comment)

[YOMO-1402]: https://emurgo.atlassian.net/browse/YOMO-1402?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[YOMO-1351]: https://emurgo.atlassian.net/browse/YOMO-1351?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[YOMO-1353]: https://emurgo.atlassian.net/browse/YOMO-1353?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/wallet-mobile/src/features/SetupWallet/SetupWalletNavigator.tsx
Comment (banklesss) [inline apps/wallet-mobile/src/features/SetupWallet/SetupWalletNavigator.tsx:1]: https://emurgo.atlassian.net/browse/YOMO-1402

forgot to add in previous pr

File: apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/MnemonicInput/MnemonicInput.tsx [hunk 1]
```diff
@@ -197,19 +208,18 @@ const MnemonicWordInput = React.forwardRef<MnemonicWordInputRef, MnemonicWordInp
...
       onClearError,
+      error,
     },
     ref,
   ) => {
     const inputRef = React.useRef<RNTextInput>(null)
     const {styles} = useStyles()
-    const [word, setWord] = React.useState(mnemonicSelectedWords[index])
-    const [error, setError] = React.useState(false)
+    const [word, setWord] = React.useState(selectedWord)
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/MnemonicInput/MnemonicInput.tsx:224]: refactor to remove duplicated state 👀

File: apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/MnemonicInput/MnemonicInput.tsx [hunk 2]
```diff
@@ -3,14 +3,14 @@ import {wordlists} from 'bip39'
...
 import {Platform, StyleSheet, Text, TextInput as RNTextInput, TouchableOpacity, View} from 'react-native'
 
-import {Spacer, useScrollView} from '../../../../components'
-import {Space} from '../../../../components/Space/Space'
-import {isEmptyString} from '../../../../utils/utils'
-import {Alert as AlertIllustration} from '../../illustrations/Alert'
-import {Check2} from '../../illustrations/Check2'
-import {MnemonicWordInputRef} from '../../useCases/RestoreWallet/RestoreWalletScreen'
-import {TextInput} from '../TextInput'
-import {useStrings} from '../useStrings'
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/MnemonicInput/MnemonicInput.tsx:13]: just moving this component to the proper folder (only used once)

File: apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx [hunk 3]
```diff
@@ -183,7 +175,7 @@ export const RestoreWalletScreen = () => {
           <StepperProgress currentStep={1} currentStepTitle={strings.stepRestoreWalletScreen} totalSteps={2} />
         </View>
 
-        <ScrollView style={styles.scroll} bounces={false}>
+        <ScrollView style={styles.scroll} bounces={false} keyboardShouldPersistTaps="always">
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx:179]: https://emurgo.atlassian.net/browse/YOMO-1351

**Comment by banklesss** [comment]
changed to draft to check one more thing

File: apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/MnemonicInput/MnemonicInput.tsx [hunk 4]
```diff
@@ -98,9 +101,11 @@ const ClearAllButton = ({onPress}: {onPress: () => void}) => {
   const {styles} = useStyles()
   const strings = useStrings()
   return (
-    <TouchableOpacity activeOpacity={0.5} style={styles.textView} onPress={onPress}>
-      <Text style={styles.clearAll}>{strings.clearAll}</Text>
-    </TouchableOpacity>
+    <View style={styles.textView}>
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/MnemonicInput/MnemonicInput.tsx:104]: the whole row was a button 👀

File: apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx [hunk 5]
```diff
@@ -107,6 +99,7 @@ export const RestoreWalletScreen = () => {
 
     if (!mnemonicWordsComplete) {
       if (isValid) setIsValidPhrase(false)
+      if (!isEmptyString(mnemonic)) setMnemonic('')
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx:102]: https://emurgo.atlassian.net/browse/YOMO-1353

**Review by stackchain — APPROVED** [review]

**Pull request merged**